views/js/qtiCreator/widgets/interactions/mediaInteraction/states/Question.js

```diff
@@ -20,153 +44,210 @@ define([
-            xmlUpdateCheat(interaction);
             widget.destroyInteraction();
             widget.renderInteraction();
         }, 1000);
 
-        var xmlUpdateCheat = function xmlUpdateCheat(interaction){
-            // xml update cheat
-            interaction.attr('responseIdentifier', interaction.attr('responseIdentifier'));
+        /**
+         * Switch to audio mode:
+         * update height and disable the field
+         */
+        var switchToAudio = function switchToAudio(){
+            isAudio = true;
+
+            $heightContainer.hide();
+            interaction.object.attr('height', defaultAudioHeight);
+        };
+
+        /**
+         * Switch to video mode:
+         * update height and enable the field
+         */
+        var switchToVideo = function switchToVideo(){
+            isAudio = false;
+            interaction.object.attr('height', defaultVideoHeight);
+            $heightContainer.show();
+        };
+
+        /**
+         * Switch mode based on file type
+         */
+        var switchMode = function switchMode(){
+            if (/audio/.test(interaction.object.attr('type'))) {
+                switchToAudio();
+            } else {
+                switchToVideo();
+            }
+        };
+
+        /**
+         * Set up the file upload component
+         */
+        var setUpUploader = function setUpUploader() {
+
+            var $src = $form.find('input[name=data]');
+            var $uploadTrigger = $form.find('.selectMediaFile');
+            var openResourceMgr = function openResourceMgr(){
+                $uploadTrigger.resourcemgr({
+                    title : __('Please select a media file (video or audio) from the resource manager. You can add files from your computer with the button "Add file(s)".'),
+                    appendContainer: options.mediaManager.appendContainer,
+                    mediaSourcesUrl: options.mediaManager.mediaSourcesUrl,
+                    browseUrl:       options.mediaManager.browseUrl,
+                    uploadUrl:       options.mediaManager.uploadUrl,
+                    deleteUrl:       options.mediaManager.deleteUrl,
+                    downloadUrl:     options.mediaManager.downloadUrl,
+                    fileExistsUrl:   options.mediaManager.fileExistsUrl,
+                    params : {
+                        uri : options.uri,
+                        lang : options.lang,
+                        filters : 'video/mp4,video/avi,video/ogv,video/mpeg,video/ogg,video/quicktime,video/webm,video/x-ms-wmv,video/x-flv,audio/mp3,audio/vnd.wav,audio/ogg,audio/vorbis,audio/webm,audio/mpeg,application/ogg,audio/aac,audio/wav,audio/flac'
+                    },
+                    pathParam : 'path',
+                    select : function(e, files){
+                        if(files && files.length){
+                            // set data field content and meybe detect and set media type here
```

Comment:
Not very important but i think there is a typo here: `maybe` instead of `meybe` ?